benchmark/largerandom/ondemand.h

```diff
@@ -28,11 +28,10 @@ simdjson_really_inline bool OnDemand::Run(const padded_string &json) {
   container.clear();
 
   auto doc = parser.iterate(json);
-  // TODO this sucks, you should be able to just say for ( ... : doc)
+  // TODO you should be able to just say for ( ... : doc)
   auto array = doc.get_array();
   for (ondemand::object point_object : array) {
-    auto point = point_object.begin();
-    container.emplace_back(my_point{(*point).value(), (*++point).value(), (*++point).value()});
```

Comment:
This is fair :)

Reply:
Right. I would argue John that this code (`my_point{(*point).value(), (*++point).value(), (*++point).value()}`) is not something you'd want to advertise... especially when your API supports something much nicer. (If it is all your API did, I'd be fine, but you wrote something that allows nicer code.)